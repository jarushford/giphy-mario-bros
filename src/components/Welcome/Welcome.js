import React from 'react'
import '../../main.scss'

export default function Welcome() {
  return (
    <section className="welcome">
      <h1 className="title">Giphy Mario Bros.</h1>
      <p className="instructions">To play this game, select your number of players. On each players turn they will be given a caption. Every other player must choose a GIF to match that caption when they are prompted to do so. Once everyone has chosen, the player whose turn it is will choose their favorite and that player will be awarded coins. Then it will move to the next player. The game starts with Player 1!</p>
      <button className="start-btn">Start</button>
    </section>
  )
}
