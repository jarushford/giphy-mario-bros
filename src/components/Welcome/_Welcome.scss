.welcome {
  width: 50%;
  margin: 0px auto;
  padding: 20px 30px;
  background: $bg;
  border-radius: 15px;
  max-width: 500px;
}

.title,
.instructions {
  margin: 10px;
  text-align: center;
}

.title {
  font-family: 'Mario';
  font-size: 2.7rem;
  color: $white;
  text-shadow: $outlinethick;
}

.instructions {
  background: $lightbg;
  padding: 10px;
  margin: 25px auto 30px;
  border-radius: 5px;
  color: $bg;
  font-family: 'Viga', sans-serif;
  font-size: 1.3rem;
  box-shadow: 0 0 35px 1px $bg;
}

.start-btn {
  @include btn();
}