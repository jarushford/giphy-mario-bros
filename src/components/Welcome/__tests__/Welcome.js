import React from 'react'
import Welcome from '../Welcome'
import { shallow } from 'enzyme'

describe('Welcome', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Welcome />)

    expect(wrapper).toMatchSnapshot()
  })
})