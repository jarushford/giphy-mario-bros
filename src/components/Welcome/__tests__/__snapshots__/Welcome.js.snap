// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Giphy Mario Bros.
        </h1>,
        <p
          className="instructions"
        >
          To play this game, select your number of players. On each players turn they will be given a caption. Every other player must choose a GIF to match that caption when they are prompted to do so. Once everyone has chosen, the player whose turn it is will choose their favorite and that player will be awarded coins. Then it will move to the next player. The game starts with Player 1!
        </p>,
        <Link
          replace={false}
          to="/select-players"
        >
          <button
            className="start-btn"
          >
            Start
          </button>
        </Link>,
      ],
      "className": "welcome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Giphy Mario Bros.",
          "className": "title",
        },
        "ref": null,
        "rendered": "Giphy Mario Bros.",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To play this game, select your number of players. On each players turn they will be given a caption. Every other player must choose a GIF to match that caption when they are prompted to do so. Once everyone has chosen, the player whose turn it is will choose their favorite and that player will be awarded coins. Then it will move to the next player. The game starts with Player 1!",
          "className": "instructions",
        },
        "ref": null,
        "rendered": "To play this game, select your number of players. On each players turn they will be given a caption. Every other player must choose a GIF to match that caption when they are prompted to do so. Once everyone has chosen, the player whose turn it is will choose their favorite and that player will be awarded coins. Then it will move to the next player. The game starts with Player 1!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="start-btn"
          >
            Start
          </button>,
          "replace": false,
          "to": "/select-players",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Start",
            "className": "start-btn",
          },
          "ref": null,
          "rendered": "Start",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Giphy Mario Bros.
          </h1>,
          <p
            className="instructions"
          >
            To play this game, select your number of players. On each players turn they will be given a caption. Every other player must choose a GIF to match that caption when they are prompted to do so. Once everyone has chosen, the player whose turn it is will choose their favorite and that player will be awarded coins. Then it will move to the next player. The game starts with Player 1!
          </p>,
          <Link
            replace={false}
            to="/select-players"
          >
            <button
              className="start-btn"
            >
              Start
            </button>
          </Link>,
        ],
        "className": "welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Giphy Mario Bros.",
            "className": "title",
          },
          "ref": null,
          "rendered": "Giphy Mario Bros.",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To play this game, select your number of players. On each players turn they will be given a caption. Every other player must choose a GIF to match that caption when they are prompted to do so. Once everyone has chosen, the player whose turn it is will choose their favorite and that player will be awarded coins. Then it will move to the next player. The game starts with Player 1!",
            "className": "instructions",
          },
          "ref": null,
          "rendered": "To play this game, select your number of players. On each players turn they will be given a caption. Every other player must choose a GIF to match that caption when they are prompted to do so. Once everyone has chosen, the player whose turn it is will choose their favorite and that player will be awarded coins. Then it will move to the next player. The game starts with Player 1!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="start-btn"
            >
              Start
            </button>,
            "replace": false,
            "to": "/select-players",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Start",
              "className": "start-btn",
            },
            "ref": null,
            "rendered": "Start",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
